instrumentation/ratpack/ratpack-1.7/library/src/main/java/io/opentelemetry/instrumentation/ratpack/v1_7/RatpackTelemetryBuilder.java

```diff
@@ -39,38 +26,20 @@ public final class RatpackTelemetryBuilder {
 
   private static final String INSTRUMENTATION_NAME = "io.opentelemetry.ratpack-1.7";
 
-  private final OpenTelemetry openTelemetry;
-
-  private final List<AttributesExtractor<? super Request, ? super Response>> additionalExtractors =
-      new ArrayList<>();
-  private final HttpClientAttributesExtractorBuilder<RequestSpec, HttpResponse>
-      httpClientAttributesExtractorBuilder =
-          HttpClientAttributesExtractor.builder(RatpackHttpClientAttributesGetter.INSTANCE);
-  private final HttpServerAttributesExtractorBuilder<Request, Response>
-      httpServerAttributesExtractorBuilder =
-          HttpServerAttributesExtractor.builder(RatpackHttpAttributesGetter.INSTANCE);
-
-  private final HttpSpanNameExtractorBuilder<RequestSpec> httpClientSpanNameExtractorBuilder =
-      HttpSpanNameExtractor.builder(RatpackHttpClientAttributesGetter.INSTANCE);
-  private final HttpSpanNameExtractorBuilder<Request> httpServerSpanNameExtractorBuilder =
-      HttpSpanNameExtractor.builder(RatpackHttpAttributesGetter.INSTANCE);
-
-  private Function<SpanNameExtractor<RequestSpec>, ? extends SpanNameExtractor<? super RequestSpec>>
-      clientSpanNameExtractorTransformer = Function.identity();
-  private Function<SpanNameExtractor<Request>, ? extends SpanNameExtractor<? super Request>>
-      serverSpanNameExtractorTransformer = Function.identity();
-
-  private final HttpServerRouteBuilder<Request> httpServerRouteBuilder =
-      HttpServerRoute.builder(RatpackHttpAttributesGetter.INSTANCE);
-
-  private final List<AttributesExtractor<? super RequestSpec, ? super HttpResponse>>
-      additionalHttpClientExtractors = new ArrayList<>();
-
-  private boolean emitExperimentalHttpClientMetrics = false;
-  private boolean emitExperimentalHttpServerMetrics = false;
+  private final DefaultHttpClientInstrumenterBuilder<RequestSpec, HttpResponse> clientBuilder;
+  private final DefaultHttpServerInstrumenterBuilder<Request, Response> serverBuilder;
 
   RatpackTelemetryBuilder(OpenTelemetry openTelemetry) {
-    this.openTelemetry = openTelemetry;
+    clientBuilder =
+        new DefaultHttpClientInstrumenterBuilder<>(
+                INSTRUMENTATION_NAME, openTelemetry, RatpackHttpClientAttributesGetter.INSTANCE)
+            .setHeaderSetter(RequestHeaderSetter.INSTANCE);
```

Comment:
Probably not really a part of this PR but `setHeaderSetter` feels a bit weird (also why does this method accept `null`?). Server builder always takes `TextMapGetter` as constructor argument. Why not add a second constructor for `DefaultHttpClientInstrumenterBuilder` that also takes `TextMapSetter`? Or have an overload of build takes it to match the underlying instrumenter builder?

Reply:
> Why not add a second constructor for `DefaultHttpClientInstrumenterBuilder` that also takes `TextMapSetter`? 

great idea - changed for both server and client